dispatcher/MongoConnect.py

```diff
@@ -220,69 +218,38 @@ def aggregate_status(self):
                     self.log.debug(f'Rate calculation ran into {type(e)}: {e}')
                     pass
 
-                try:
-                    status = DAQ_STATUS(doc['status'])
-                    if self.is_timeout(doc, now_time):
-                        status = DAQ_STATUS.TIMEOUT
-                except Exception as e:
-                    self.log.debug(f'Ran into {type(e)}, daq is in timeout. {e}')
-                    status = DAQ_STATUS.UNKNOWN
-
+                status = self.extract_status(doc, now_time)
                 statuses[doc['host']] = status
+                phys_stat[phys_det].append(status)
 
             for doc in self.latest_status[detector]['controller'].values():
                 phys_det = self.host_config[doc['host']]
-                try:
-                    status = DAQ_STATUS(doc['status'])
-
-                    if self.is_timeout(doc, now_time):
-                        status = DAQ_STATUS.TIMEOUT
-                except Exception as e:
-                    self.log.debug(f'Setting status to unknown because of {type(e)}: {e}')
-                    status = DAQ_STATUS.UNKNOWN
-
+                status = self.extract_status(doc, now_time)
                 statuses[doc['host']] = status
+                doc['status'] = status
                 modes.append(doc.get('mode', 'none'))
                 run_nums.append(doc.get('number', None))
                 aggstat[phys_det]['status'] = status
                 aggstat[phys_det]['mode'] = modes[-1]
                 aggstat[phys_det]['number'] = run_nums[-1]
+                phys_stat[phys_det].append(status)
```

Comment:
Here you would also need the 
```python
                if phys_det not in phys_stat:
                    phys_stat[phys_det] = []
```
unless you use the defaultdict, right?

Reply:
This is L210, I guess we could also add it after L226 but there are cleaner solutions.